date.libsonnet

```diff
@@ -55,4 +55,131 @@ local d = import 'doc-util/main.libsonnet';
+    assert std.type(sep) == 'string' : 'parseSeparatedNumbers() only operates on string separators, got %s' % std.type(sep);
+    assert std.type(names) == 'array' : 'parseSeparatedNumbers() only operates on arrays of names, got input %s' % std.type(names);
+
+    local parts = std.split(input, sep);
+    assert std.length(parts) == std.length(names) : 'expected %(expected)d parts separated by %(sep)s in %(format)s formatted input "%(input)s", but got %(got)d' % {
+      expected: std.length(names),
+      sep: sep,
+      format: std.join(sep, names),
+      input: input,
+      got: std.length(parts),
+    };
+
+    {
+      [names[i]]:
+        // Fail with meaningful message if not numeric, otherwise it will be a hell to debug.
+        assert isNumeric(parts[i]) : '%(name)%s part "%(part)s" of %(format)s of input "%(input)s" is not numeric' % {
+          name: names[i],
+          part: parts[i],
+          format: std.join(sep, names),
+          input: input,
+        };
+        std.parseInt(parts[i])
+      for i in std.range(0, std.length(parts) - 1)
+    }
+  ),
+
+  // stringContains is a helper function to check whether a string contains a given substring.
+  local stringContains(haystack, needle) = std.length(std.findSubstr(needle, haystack)) > 0,
+
+  '#parseRFC3339': d.fn(
+    |||
+      `parseRFC3339` parses an RFC3339-formatted date & time string into an object containing the 'year', 'month', 'day', 'hour', 'minute' and 'second fields.
+      This is a limited implementation that does not support timezones (so it requires an UTC input ending in 'Z' or 'z') nor sub-second precision.
+      The returned object has a `toUnixTimestamp()` method that can be used to obtain the unix timestamp of the parsed date.
+    |||,
+    [
+      d.arg('input', d.T.string),
+    ],
+  ),
+  parseRFC3339(input)::
+    // Basic input type check.
+    assert std.type(input) == 'string' : 'parseRFC3339() only operates on string inputs, got %s' % std.type(input);
+
+    // Sub-second precision isn't implemented yet, warn the user about that instead of returning wrong results.
+    assert !stringContains(input, '.') : 'the provided RFC3339 input "%s" has a dot, most likely representing a sub-second precision, but this function does not support that' % input;
+
+    // We don't support timezones, so string should end with 'Z' or 'z'.
+    assert std.endsWith(input, 'Z') || std.endsWith(input, 'z') : 'the provided RFC3339 "%s" should end with "Z" or "z". This implementation does not currently support timezones' % input;
+
+    // RFC3339 can separate date and time using 'T', 't' or ' '.
+    // Find out which one it is and use it.
+    local sep =
+      if stringContains(input, 'T') then 'T'
+      else if stringContains(input, 't') then 't'
+      else if stringContains(input, ' ') then ' '
+      else error 'the provided RFC3339 input "%s" should contain either a "T", or a "t" or space " " as a separator for date and time parts' % input;
```

Comment:
@Duologic as you asked in https://github.com/grafana/jsonnet-libs/pull/1095, I added support for lowercase t/z and also for space separator.